controllers/apps/components/internal/component_base_stateful_hscale_backup.go

```diff
@@ -0,0 +1,649 @@
+package internal
+
+import (
+	"fmt"
+	"strings"
+
+	snapshotv1 "github.com/kubernetes-csi/external-snapshotter/client/v6/apis/volumesnapshot/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1alpha1 "github.com/apecloud/kubeblocks/apis/apps/v1alpha1"
+	dataprotectionv1alpha1 "github.com/apecloud/kubeblocks/apis/dataprotection/v1alpha1"
+	"github.com/apecloud/kubeblocks/internal/constant"
+	"github.com/apecloud/kubeblocks/internal/controller/builder"
+	"github.com/apecloud/kubeblocks/internal/controller/component"
+	intctrlutil "github.com/apecloud/kubeblocks/internal/controllerutil"
+)
+
+type DataClone interface {
+	// Succeed check if data clone succeeded
+	Succeed() (bool, error)
+	// CloneData do clone data, return objects that need to be created
+	CloneData(innerDataClone) ([]client.Object, error)
```

Comment:
a public interface function with private type param is not making a lot sense to me.